### PR TITLE
changing entity to counter

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
following naming convention

### CODE REVIEW

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto [hunk 1]
```diff
@@ -7,26 +7,26 @@ import "google/api/annotations.proto";
...
 
-service MyServiceEntity {
+service CounterService {
     option (akkaserverless.service) = {
         type : SERVICE_TYPE_ENTITY
-        component : ".domain.MyEntity"
+        component : ".domain.Counter"
     };
     rpc SetValue(SetValueCommand) returns (google.protobuf.Empty);
-    rpc GetValue(GetValueCommand) returns (MyResult);
```
Comment (johanandren) [inline maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto:31]: Should filename(s) match as well?

Comment (octonato) [reply maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto:31]: yes, they should

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto [hunk 2]
```diff
@@ -7,26 +7,26 @@ import "google/api/annotations.proto";
 
 package $package;
 
-option java_outer_classname = "MyEntityApi";
+option java_outer_classname = "CounterApi";
 
 message SetValueCommand {
```
Comment (octonato) [inline maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto:12]: We should aim for the same commands we added for the value entity counter, eg: 

```proto
message IncreaseValue { // <4>
  string counter_id = 1 [(akkaserverless.field).entity_key = true]; // <5>
  int32 value = 2;
}

message DecreaseValue {
  string counter_id = 1 [(akkaserverless.field).entity_key = true];
  int32 value = 2;
}

message ResetValue {
  string counter_id = 1 [(akkaserverless.field).entity_key = true];
}

message GetCounter {
  string counter_id = 1 [(akkaserverless.field).entity_key = true];
}
```

and corresponding events, `ValueIncreased`, `Reset`, etc.

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto [hunk 3]
```diff
@@ -7,26 +7,26 @@ import "google/api/annotations.proto";
...
-service MyServiceEntity {
+service CounterService {
     option (akkaserverless.service) = {
         type : SERVICE_TYPE_ENTITY
-        component : ".domain.MyEntity"
+        component : ".domain.Counter"
     };
     rpc SetValue(SetValueCommand) returns (google.protobuf.Empty);
-    rpc GetValue(GetValueCommand) returns (MyResult);
+    rpc GetValue(GetValueCommand) returns (CurrentCounter);
```
Comment (octonato) [inline maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto:31]: also methods should follow the value entity variant

```proto
rpc Increase (IncreaseValue) returns (google.protobuf.Empty);
rpc Decrease (DecreaseValue) returns (google.protobuf.Empty);
rpc Reset (ResetValue) returns (google.protobuf.Empty);
rpc GetCurrentCounter (GetCounter) returns (CurrentCounter);
```

**Review by octonato — COMMENTED** [review]
@franciscolopezsancho, the sample must be closed to the value entity one. 

One thing that we did for the shopping cart, is that the external Service Api is exactly the same. This allow us to have instructions in the readme file that are exactly the same with respect to the commands we can send.

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto (hunk 1 above)
Comment (franciscolopezsancho) [reply maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto:31]: done now

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto (hunk 3 above)
Comment (franciscolopezsancho) [reply maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto]: done now

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto (hunk 2 above)
Comment (franciscolopezsancho) [reply maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/myentity_api.proto]: done now

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    state: "CounterState"
+    events: ["ValueIncreased","ValueDecreased","ValueReset"]
+};
+
+message CounterState {
+    int32 value = 1;
+}
+
+message ValueIncreased {
+    string entity_id = 1 [(akkaserverless.field).entity_key = true];
```
Comment (octonato) [inline maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto:21]:
```suggestion
    string counter_id = 1 [(akkaserverless.field).entity_key = true];
```

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+message ValueDecreased {
+    string entity_id = 1 [(akkaserverless.field).entity_key = true];
```
Comment (octonato) [inline maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto:25]:
```suggestion
    string counter_id = 1 [(akkaserverless.field).entity_key = true];
```

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+message ValueReset {
+    string entity_id = 1 [(akkaserverless.field).entity_key = true];
```
Comment (octonato) [inline maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto:29]:
```suggestion
    string counter_id = 1 [(akkaserverless.field).entity_key = true];
```

**Review by octonato — APPROVED** [review]
Looks good @franciscolopezsancho. Thanks. 

I left one small suggestion. Just to align the id field with the commands. 

Otherwise, good to merge.

File: maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto (hunk 6 above)
Comment (franciscolopezsancho) [reply maven-java/maven-archetype-akkasls-event-sourced-entity/src/main/resources/archetype-resources/src/main/proto/counter_domain.proto]: thanks!

**Pull request merged**